netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -104,39 +103,42 @@ public void ping(PingCallback callback, Executor executor) {
   }
 
   @Override
-  public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers,
-      ClientStreamListener listener) {
+  public ClientStream newStream(final MethodDescriptor<?, ?> method, final Metadata headers) {
     Preconditions.checkNotNull(method, "method");
     Preconditions.checkNotNull(headers, "headers");
-    Preconditions.checkNotNull(listener, "listener");
-
-    // Create the stream.
-    final NettyClientStream stream = new NettyClientStream(listener, channel, handler,
-        maxMessageSize);
 
     // Convert the headers into Netty HTTP/2 headers.
     AsciiString defaultPath = new AsciiString("/" + method.getFullMethodName());
-    AsciiString defaultAuthority  = new AsciiString(headers.containsKey(AUTHORITY_KEY)
+    AsciiString defaultAuthority = new AsciiString(headers.containsKey(AUTHORITY_KEY)
         ? headers.get(AUTHORITY_KEY) : authority);
     headers.removeAll(AUTHORITY_KEY);
-    Http2Headers http2Headers = Utils.convertClientHeaders(headers, negotiationHandler.scheme(),
-        defaultPath, defaultAuthority);
+    final Http2Headers http2Headers = Utils.convertClientHeaders(
+        headers, negotiationHandler.scheme(), defaultPath, defaultAuthority);
+
+    class StartCallback implements Runnable {
```

Comment:
Similar to `InProcessTransport`, can we move the following work into `NettyClientStream` and get rid of this ad-hoc callback?